bin/character-gen.js

```diff
@@ -10,7 +10,7 @@ const isInlineBlank = c => c === '\t' || c === ' ' || c === '\r'
 const isSign = c => c === '-' || c === '+'
 // See https://developer.mozilla.org/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/RegExp
 const unicodeBlanks = '\u00a0\u1680\u180e\u2000\u2001\u2002\u2003\u2004\u2005\u2006\u2007\u2008\u2009\u200a\u2028\u2029\u202f\u205f\u3000'
-const unicodePunctuations = '“”'
+const unicodePunctuation = '“”'
```

Comment:
[punctuation](https://www.dictionary.com/browse/punctuation) here (3) is [`punctuation marks`](https://www.dictionary.com/browse/punctuation-mark) -- it already refers to punctuation marks -- you wouldn't write "one punctuation" although you could write "one punctuation mark".